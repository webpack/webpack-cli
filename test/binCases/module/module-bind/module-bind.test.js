"use strict";

const { run, extractSummary } = require("../../../testUtils");

test("module-bind", () => {
	const { code, stdout, stderr } = run(__dirname, [
		"--entry",
		"./index.js",
		"--config",
		"./webpack.config.js",
		"--output-filename",
		"[name].js",
		"--output-chunk-filename",
		"[id].chunk.js",
		"--target",
		"async-node",
		"--module-bind-pre",
		"pre=./pre-loader",
		"--module-bind-post",
		"post=./post-loader"
	]);
	expect(code).toBe(0);

	const summary = extractSummary(stdout);

	expect(summary).toEqual(expect.anything());
	expect(summary).toContain("pre-loaded pre");
	expect(summary).toContain("post-loaded post");
	expect(stderr).toHaveLength(0);
});
