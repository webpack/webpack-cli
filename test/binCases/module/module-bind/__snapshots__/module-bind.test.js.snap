// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module-bind 1`] = `
"pre-loaded pre
post-loaded post
  Asset       Size  Chunks             Chunk Names
null.js  996 bytes       0  [emitted]  null
Entrypoint null = null.js
[0] ./index.js 128 bytes {0} [built]
[1] ./file.pre 4 bytes {0} [built]
[2] ./file.post 5 bytes {0} [built]

WARNING in configuration
The 'mode' option has not been set, webpack will fallback to 'production' for this value. Set 'mode' option to 'development' or 'production' to enable defaults for each environment.
You can also set it to 'none' to disable any default behavior. Learn more: https://webpack.js.org/concepts/mode/"
`;
